src/auth/auth.service.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotAcceptableException,
+} from '@nestjs/common';
+import { RegisterDto } from './dto/register.dto';
+import { PrismaService } from 'src/prisma.service';
+import { LoginDto } from './dto/login.dto';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private primaService: PrismaService,
+    private jwtService: JwtService,
+  ) {}
+
+  async validateUser(email: string, password: string): Promise<any> {
+    const user = await this.primaService.user.findUnique({ where: { email } });
+    if (!user) return null;
+    const passwordValid = await this.comparePassword(password, user.password);
+    if (!user) {
+      throw new NotAcceptableException();
+    }
+    if (user && passwordValid) {
+      return user;
+    }
+    return null;
+  }
+
+  async register(registerData: RegisterDto): Promise<string> {
+    try {
+      const hashedPassword = await this.hashPassword(registerData.password);
+      const data = await this.primaService.user.create({
+        data: {
+          email: registerData.email,
+          password: hashedPassword,
+        },
+      });
+      const token = this.jwtService.sign(
+        {
+          id: data.id,
+          email: data.email,
+        },
+        {
+          secret: process.env.SECRET_JWT_TOKEN,
+        },
```

Comment:
I think we don't need to add secret key here, because in the AuthModule you registered the secret